python/GafferUI/Menu.py

```diff
@@ -368,6 +375,11 @@ def __buildAction( self, item, name, parent ) :
 		if item.description :
 			qtAction.setStatusTip( item.description )
 
+		if hasattr( item, 'enter' ) :
+			parent.enterCallbacks[ name ] = item.enter
```

Comment:
Why do we need to store these separately, and why on the parent? Why can't we retrieve them from the `qtAction.__item` that we've already stashed on line 322/327?

Reply:
https://github.com/GafferHQ/gaffer/pull/3360/commits/34f015d55ce9017418e74dc1208ce6077a69a2b2 sorts this all out. Long story short, randomly added attributes were being dropped from the object. Deriving our own action sorted this.